libs/@blockprotocol/graph/src/types/ontology/property-type.ts

```diff
@@ -12,3 +19,29 @@ export type PropertyTypeWithMetadata = {
   schema: PropertyType;
   metadata: OntologyElementMetadata;
 };
+
+export type AggregatePropertyTypesData = {
+  graphResolveDepths?: Partial<
+    Pick<
+      Subgraph<true>["depths"],
+      "constrainsValuesOn" | "constrainsPropertiesOn"
+    >
+  >;
+};
+
+export type AggregatePropertyTypesResult = {
+  results: Subgraph<true, PropertyTypeRootType>;
+};
+
+export type GetPropertyTypeData = {
+  propertyTypeId: VersionedUri;
+};
+
+export type CreatePropertyTypeData = {
+  propertyType: PropertyType;
+};
+
+export type UpdatePropertyTypeData = {
+  propertyTypeId: VersionedUri;
+  propertyType: PropertyType;
+};
```

Comment:
Similar comment as above. A caller cannot determine the type ID, and generally the EA should be responsible for ensuring the version was incremented appropriately.

Reply:
Done in https://github.com/blockprotocol/blockprotocol/pull/946/commits/086f506ad7e3054eed4c6896363cce4d53a06145

Although I think a legitimate alternative for updates is to allow it (to save callers omitting it, since some might send it anyway).